<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">



<title>Introduction to metaMint</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to metaMint</h1>



<p>This document introduces to you how to use the package <strong>metaMint</strong> for joint estimation of metabolite and microbial interaction network.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw">library</span>(metaMint)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">library</span>(dplyr)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw">library</span>(GGally)</a></code></pre></div>
<div id="data" class="section level2">
<h2>Data</h2>
<p>We use an example data set from <span class="citation">McMillan et al. (2015)</span> that has 51 bacterial taxa and 128 metabolites measured over 131 samples. Taxonomic counts were obtained using 16S rRNA gene sequencing, whereas metabolite profiling was done by gas chromatography-mass spectrometry (GC-MS).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw">data</span>(BV, <span class="dt">package =</span> <span class="st">&quot;metaMint&quot;</span>)</a></code></pre></div>
<p>Before calling the package, it is important to preprocess the taxonomic count data which is subject to the total library size constraint. A common practice is to apply the centered log-ratio (clr) transformation. Briefly, for a vector <span class="math inline">\(r\)</span> with <span class="math inline">\(p\)</span> components, let <span class="math inline">\(g(r)=(\Pi_{j=1}^p)^{1/p}\)</span> denote its geometric mean. The clr of <span class="math inline">\(r\)</span> is defined as <span class="math display">\[
{\rm clr}(r) = \left( \log \frac{r_1}{g(r)},\ldots,\log \frac{r_p}{g(r)}\right)^\intercal.
\]</span> It is easy to see that the clr transformation does not allow zero counts. To address this issue, a small pseudocount of 0.5 or 1 is often added to all counts before applying the transformation. However, adding pseudocounts may unfairly bias rare species.</p>
<p>To avoid introducing artificial bias, we use instead the modified-clr (mclr) transformation. The mclr transforms the non-zero counts with the usual clr and shifts all transformed values to be strictly positive. Details on how mclr compares with clr transformation can be found in <span class="citation">Yoon, Gaynanova, and Müller (2019)</span>. The following Figure illustrates the different marginal distributions one gets with clr and mclr transformation. The mclr transformed value can be modeled using a censored normal distribution, which provides the basis for our method.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">OTUs_clr &lt;-<span class="st"> </span><span class="kw">apply</span>(OTUs,<span class="dv">2</span>,compositions<span class="op">::</span>clr)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">OTUs_mclr &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">clr.epsilon</span>(<span class="kw">t</span>(OTUs)))</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="kw">ggpairs</span>(<span class="kw">data.frame</span>(<span class="dt">clr=</span>OTUs_clr[<span class="dv">22</span>,],</a>
<a class="sourceLine" id="cb3-5" data-line-number="5">                   <span class="dt">mclr=</span>OTUs_mclr[<span class="dv">22</span>,]),</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">        <span class="dt">diag =</span> <span class="kw">list</span>(<span class="dt">continuous =</span> <span class="st">&quot;barDiag&quot;</span>,<span class="dt">binwidth =</span> <span class="fl">0.5</span>))</a></code></pre></div>
</div>
<div id="estimation" class="section level2">
<h2>Estimation</h2>
<p>With preprocessed data, we are now ready to show how to use the main functions in <strong>metaMint</strong>. One can use the function <code>cggm.corr</code> to estimate the correlations among bacterial taxa.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">est_Cor &lt;-<span class="st"> </span><span class="kw">cggm.corr</span>(<span class="kw">t</span>(OTUs_mclr))</a></code></pre></div>
<p>Alternatively, if the interest is to learn the conditional association (as defined by partial correlations) among bacterial taxa, one can use the function <code>cggm.pcorr</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">est_InvCor &lt;-<span class="st"> </span><span class="kw">cggm.pcorr</span>(<span class="kw">t</span>(OTUs_mclr))</a></code></pre></div>
<p>The model underlying <strong>metaMint</strong> allows for both continuous and zero-inflated variables. Therefore, to jointly estimate the metabolite and microbial interaction network, one only needs to expand the input to include both types of measurements.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1">dat_combined &lt;-<span class="st"> </span><span class="kw">t</span>(<span class="kw">rbind</span>(OTUs_mclr,metabs))</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">est_InvCor_combined &lt;-<span class="st"> </span><span class="kw">cggm.pcorr</span>(dat_combined)</a></code></pre></div>
</div>
<div id="model-selection" class="section level2">
<h2>Model Selection</h2>
<p>The function <code>cggm.stars</code> provides a way for selecting the optimal regularization parameter that controls the sparsity of the estimated partial correlation network. It is based on the idea of stability selection <span class="citation">(Liu, Roeder, and Wasserman 2010)</span>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1">est_InvCor &lt;-<span class="st"> </span><span class="kw">cggm.pcorr</span>(<span class="kw">t</span>(OTUs_mclr))</a>
<a class="sourceLine" id="cb7-2" data-line-number="2">stars_obj &lt;-<span class="st"> </span><span class="kw">cggm.stars</span>(est_InvCor,<span class="dt">stars.thresh =</span> <span class="fl">0.1</span>, <span class="dt">rep.num =</span> <span class="dv">20</span>)</a></code></pre></div>
</div>
<div id="references" class="section level2 unnumbered">
<h2>References</h2>
<div id="refs" class="references">
<div id="ref-liu2010stability">
<p>Liu, Han, Kathryn Roeder, and Larry Wasserman. 2010. “Stability Approach to Regularization Selection (Stars) for High Dimensional Graphical Models.” In <em>Advances in Neural Information Processing Systems</em>, 1432–40.</p>
</div>
<div id="ref-mcmillan2015multi">
<p>McMillan, Amy, Stephen Rulisa, Mark Sumarah, Jean M Macklaim, Justin Renaud, Jordan E Bisanz, Gregory B Gloor, and Gregor Reid. 2015. “A Multi-Platform Metabolomics Approach Identifies Highly Specific Biomarkers of Bacterial Diversity in the Vagina of Pregnant and Non-Pregnant Women.” <em>Scientific Reports</em> 5. Nature Publishing Group: 14174.</p>
</div>
<div id="ref-yoon2019microbial">
<p>Yoon, Grace, Irina Gaynanova, and Christian L Müller. 2019. “Microbial Networks in Spring-Semi-Parametric Rank-Based Correlation and Partial Correlation Estimation for Quantitative Microbiome Data.” <em>Frontiers in Genetics</em> 10. Frontiers Media SA.</p>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
